lib/mix/lib/mix/tasks/archive.install.ex

```diff
@@ -113,7 +113,7 @@ defmodule Mix.Tasks.Archive.Install do
     remove_previous_versions(previous)
 
     File.mkdir_p!(dir_dest)
-    {:ok, _} = :zip.extract(contents, cwd: dir_dest)
+    {:ok, _} = :zip.extract(contents, cwd: to_charlist(dir_dest))
```

Comment:
Erlang expects this to be a charlist but it seems to be working for binaries right now? Not sure if it's worth looking into.

Reply:
I will confirm to the specs just in case, so this change is good IMO.